### DISCUSSION TITLE
I could only send Bluetooth signals consistently when I connect the 'sender' hub to the computer?

### DISCUSSION THREAD

**Discussion by Potato-eater** [pybricks/support#1149] [Q&A] [answered]
here is my code:

```python
from pybricks.hubs import PrimeHub
from pybricks.pupdevices import Motor, ColorSensor, UltrasonicSensor, ForceSensor
from pybricks.parameters import Button, Color, Direction, Port, Side, Stop
from pybricks.robotics import DriveBase
from pybricks.tools import wait, StopWatch

hub = PrimeHub(broadcast_channel=10)
mode = 0

while (True):
    print('') # the problem
    if (mode == 0):
        direction = 0
        x = 0
        y = 0
        t = hub.imu.tilt()
        if (t[1] > 30):
            y = 1
        elif (t[1] < -30):
            y = -1
        
        if (t[0] > 30):
            x = -1
        if (t[0] < -30):
            x = 1
        hub.ble.broadcast(x, y)
```

this code is for broadcasting signal to the other hub, but I realized that I can only broadcast this consistently if I connect this to my computer, and print something. otherwise, it would only send every two seconds, for about 5 pulses instead of sending signal all the time. Is there a way to make the robot work more independently? I don't want to always connect this hub to my computer when I run it.

**Comment by laurensvalk** [comment]
That sounds like a bug, thanks for reporting. It is expected to work with and without the PC connected.

**Reply by laurensvalk** [reply, marked as answer]
does it help if you add a `wait(10)` in your loop?

**Reply by Potato-eater** [reply]
yes, it does get better.

**Reply by laurensvalk** [reply]
Does it resolve all delays?

**Reply by Potato-eater** [reply]
yes, I think so.
Can you explain why putting ```wait(10)``` would fix this problem?

**Reply by laurensvalk** [reply]
In your original program, `hub.ble.broadcast(x, y)` is being called in a very fast loop, which may run thousands of times per second, or more. The Bluetooth chip probably can't keep up with that and gets confused about what to actually send.

The reason it works while the computer is connected, is likely because of the `print` statement, which introduces a bit of delay. That is why I suggested a small `wait` to achieve the same thing.

Thanks for bringing this up though! We should probably add a short delay internally if data was recently set, so that scripts like your original code will work without users having to worry about it.

**Reply by Potato-eater** [reply]
Thank you for the explanation!
